src/vc/vc.spec.ts

```diff
@@ -386,4 +386,113 @@ describe('VC Resolver', function () {
+      const vcKey = await runtime.vc.generateKey();
+      expect(vcKey).to.be.string;
+    });
+
+    it('does not allow the current runtime to generate same keys using vc api', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      expect(vcKey).to.be.string;
+
+      const vcKey2 = await runtime.vc.generateKey();
+      expect(vcKey2).to.be.string;
+      expect(vcKey).to.be.not.eq(vcKey2);
+    });
+
+    it('allows to store and retrieve a sharable vc from the same account', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      encryptionInfo = { key: vcKey };
+      expect(vcKey).to.be.string;
+
+      const storedVcDoc = await runtime.vc.storeVc(minimalValidVcData,
+        encryptionInfo);
+      const fetchedVcDoc = await runtime.vc.getVc(storedVcDoc.id.replace('vc:evan:testcore:', ''),
+        encryptionInfo);
+
+      expect(storedVcDoc).to.be.deep.eq(fetchedVcDoc);
+    });
+
+    it('does not allow to retrieve a sharable vc with wrong key', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      const vcKey2 = await runtime.vc.generateKey();
+      expect(vcKey).to.be.string;
+
+      encryptionInfo = { key: vcKey };
+      const storedVcDoc = await runtime.vc.storeVc(minimalValidVcData,
+        encryptionInfo);
+      encryptionInfo = { key: vcKey2 };
+      const fetchedVcDoc = runtime.vc.getVc(storedVcDoc.id.replace('vc:evan:testcore:', ''),
+        encryptionInfo);
+
+      await expect(Error).to.exist;
+      await expect(fetchedVcDoc).to.be.rejectedWith('Incorrect key: decryption failed');
+    });
+
+
+    it('allows to store an sharable vc and retrieve it from different account', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      encryptionInfo = { key: vcKey };
+      expect(vcKey).to.be.string;
+
+      const otherRuntime = await TestUtils.getRuntime(accounts[1], null, { useIdentity: true });
+      const storedVcDoc = await runtime.vc.storeVc(minimalValidVcData,
+        encryptionInfo);
+      const fetchedVcDoc = await otherRuntime.vc.getVc(storedVcDoc.id.replace('vc:evan:testcore:', ''),
+        encryptionInfo);
+
+      expect(storedVcDoc).to.be.deep.eq(fetchedVcDoc);
+    });
+
+    it('does not allow to retrieve sharable VC with wrong key from a diffirent account', async () => {
+      const vcKey = await runtime.vc.generateKey();
+      const vcKey2 = await runtime.vc.generateKey();
+      encryptionInfo = { key: vcKey };
+      expect(vcKey).to.be.string;
+
+      const otherRuntime = await TestUtils.getRuntime(accounts[1], null, { useIdentity: true });
+      const storedVcDoc = await runtime.vc.storeVc(minimalValidVcData,
+        encryptionInfo);
+      encryptionInfo = { key: vcKey2 };
+      const fetchedVcDoc = otherRuntime.vc.getVc(storedVcDoc.id.replace('vc:evan:testcore:', ''),
+        encryptionInfo);
+
+      await expect(Error).to.exist;
```

Comment:
Does this actually do anything? I think this always resolves to true, no?

Reply:
In this case it is the same as previous case. the only difference is now we try to retrieve VC from a different runtime and using the wrong key. This should result in an Error for the test to pass.